### PR TITLE
build: introduce var `LOCALBASE` for FreeBSD

### LINKED ISSUE #11380: USE_SYSTEM_SUITESPARSE=1 sets wrong include path for suitesparse on FreeBSD
In deps/Makefile the include path for suitesparse is set as `SUITESPARSE_INC = -I /usr/include/suitesparse` (Line 1555).
Which fails in FreeBSD as packages are installed into /usr/local (hence includes in /usr/local/includes) per default and possibly somewhere else. Also on Linux e.g. if you manually install suitesparse from tarball.

For completeness, the error message:
    root@101amd64-system:/wrkdirs/julia # gmake CC=gcc46 CXX=g++46 FC=gfortran46
    SuiteSparse_wrapper.c:2:21: fatal error: cholmod.h: No such file or directory
compilation terminated.
    Makefile:1564: recipe for target
'/wrkdirs/julia/usr/lib/libsuitesparse_wrapper.so' failed
    gmake[1]: **\* [/wrkdirs/julia/usr/lib/libsuitesparse_wrapper.so] Error 1
    Makefile:49: recipe for target 'julia-deps' failed
    gmake: **\* [julia-deps] Error 2

### PR DESCRIPTION
**Pull request by iblislin**
fix #11380 

----
I think it's time for adding ci for FreeBSD.
I also tried Buildbot out: https://github.com/iblis17/julia-fbsd-buildbot
If anyone is interested in this, please take a look at [my personal buildbot](http://140.113.31.207:8010/).

### CODE REVIEW

File: src/Makefile [hunk 1]
```diff
@@ -21,6 +21,7 @@ FLAGS := \
 	-D_GNU_SOURCE -I$(BUILDDIR) -I$(SRCDIR) \
 	-I$(SRCDIR)/flisp -I$(SRCDIR)/support \
 	-I$(LIBUV_INC) -I$(build_includedir) -DLIBRARY_EXPORTS \
+	-I$(LOCALBASE)/include \
```
Comment (tkelman) [inline src/Makefile]: is `/usr/include` implicitly on the include path usually?

**Comment by tkelman** [comment]
Not bad. We have a buildbot instance at build.julialang.org but not sure if we have a sparse VM handy to run freebsd on. If you're volunteering resources and would be willing to keep an eye on it, we could add a webhook if needed.

File: src/Makefile (hunk 1 above)
Comment (iblis17) [inline src/Makefile]: yes. But for freebsd, `$(LOCALBASE)/include` is `/usr/local/include`. `/usr/local/include` is not included by default

Comment (vchuravy) [inline src/Makefile]: On my system it is:

```
`gcc -print-prog-name=cc1plus` -v
ignoring nonexistent directory "/usr/lib/gcc/x86_64-pc-linux-gnu/6.3.1/../../../../x86_64-pc-linux-gnu/include"
#include "..." search starts here:
#include <...> search starts here:
 /usr/lib/gcc/x86_64-pc-linux-gnu/6.3.1/../../../../include/c++/6.3.1
 /usr/lib/gcc/x86_64-pc-linux-gnu/6.3.1/../../../../include/c++/6.3.1/x86_64-pc-linux-gnu
 /usr/lib/gcc/x86_64-pc-linux-gnu/6.3.1/../../../../include/c++/6.3.1/backward
 /usr/lib/gcc/x86_64-pc-linux-gnu/6.3.1/include
 /usr/local/include
 /usr/lib/gcc/x86_64-pc-linux-gnu/6.3.1/include-fixed
 /usr/include
End of search list.
```

So both `/usr/include` and `/usr/local/include` are on the default list, but I don't think there is anything requiring it to be, besides convention.

Comment (iblis17) [inline src/Makefile]: On Freebsd:
```
└─[iblis@ionic]% clang -E  - -v < /dev/null
FreeBSD clang version 3.9.1 (tags/RELEASE_391/final 289601) (based on LLVM 3.9.1)
Target: x86_64-unknown-freebsd12.0
Thread model: posix
InstalledDir: /usr/bin
 "/usr/bin/clang" -cc1 -triple x86_64-unknown-freebsd12.0 -E -disable-free -main-file-name - -mrelocation-model static -mthread-model posix -mdisable-fp-elim -masm-verbose -mconstructor-aliases -munwind-tables -target-cpu x86-64 -v -dwarf-column-info -debugger-tuning=gdb -resource-dir /usr/bin/../lib/clang/3.9.1 -fdebug-compilation-dir /home/iblis/git/buildbot/master -ferror-limit 19 -fmessage-length 111 -fobjc-runtime=gnustep -fdiagnostics-show-option -fcolor-diagnostics -o - -x c -
clang -cc1 version 3.9.1 based upon LLVM 3.9.1 default target x86_64-unknown-freebsd12.0
#include "..." search starts here:
#include <...> search starts here:
 /usr/bin/../lib/clang/3.9.1/include
 /usr/include
End of search list.
# 1 "<stdin>"
# 1 "<built-in>" 1
# 1 "<built-in>" 3
# 315 "<built-in>" 3
# 1 "<command line>" 1
# 1 "<built-in>" 2
# 1 "<stdin>" 2
```
Add a condiction for FreeBSD in make?

**Comment by iblis17** [comment]
>If you're volunteering resources and would be willing to keep an eye on it, we could add a webhook if needed.

Sure, I can help 😄

File: src/Makefile (hunk 1 above)
Comment (tkelman) [inline src/Makefile]: ah yeah see the appveyor failure, unconditionally adding this is bad for cases when you're cross-compiling

Comment (iblis17) [inline src/Makefile]: ok, I added a condition.

**Review by tkelman — APPROVED** [review]
guess if you're cross-compiling from freebsd then you better set LOCALBASE

**Pull request merged**